pinot-core/src/main/java/com/linkedin/pinot/core/startree/v2/function/StarTreeAggregationFunction.java

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright (C) 2014-2018 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.core.startree.v2.function;
+
+import com.linkedin.pinot.common.data.FieldSpec.DataType;
+import com.linkedin.pinot.core.query.aggregation.function.AggregationFunctionType;
+import java.io.IOException;
+
+
+/**
+ * Aggregation function for star-tree to aggregate on raw values as well as aggregated values.
+ *
+ * @param <R> Type of the raw value (non-aggregated)
+ * @param <A> Type of the aggregated value
+ */
+public interface StarTreeAggregationFunction<R, A> {
```

Comment:
Using AggregationFunction suffix makes it a bit confusing with the aggregation functions used during query execution time. Since this is completely different from that, may be using a different suffix like 'Aggregator' might help avoid that confusion.

Reply:
Will rename to **StarTreeValueAggregator**